phpunit/functional/Item_SoftwareLicenseTest.php

```diff
@@ -269,17 +261,314 @@ public function testCountLicenses()
+        $cSoftwareLicense = new Item_SoftwareLicense();
         $this->assertCount(5, $cSoftwareLicense->rawSearchOptions());
     }
+
+    /**
+     * Test that users are counted correctly in countForLicense
+     */
+    public function testUserCountingInLicense()
+    {
+        $this->login();
+
+        // Create a software
+        $software_id = $this->createItem(\Software::class, [
+            'name' => 'Test software for license counting',
+            'entities_id' => getItemByTypeName('Entity', '_test_root_entity', true),
+        ])->getID();
+
+        // Create a license with limited number
+        $license_id = $this->createItem(\SoftwareLicense::class, [
+            'name' => 'Test license with limit',
+            'softwares_id' => $software_id,
+            'number' => 2, // Limit to 2 installations
+            'entities_id' => getItemByTypeName('Entity', '_test_root_entity', true),
+        ])->getID();
+
+        // Add a computer to this license
+        $computer = getItemByTypeName('Computer', '_test_pc01');
+        $item_license_id = $this->createItem(Item_SoftwareLicense::class, [
+            'softwarelicenses_id' => $license_id,
+            'items_id' => $computer->getID(),
+            'itemtype' => 'Computer',
+        ])->getID();
+
+        // process count should be 1
+        $count = Item_SoftwareLicense::countForLicense($license_id);
+        $this->assertEquals(1, $count);
+
+        // Add a user to this license
+        $user = getItemByTypeName('User', TU_USER);
+        $user_license_id = $this->createItem(\SoftwareLicense_User::class, [
+            'softwarelicenses_id' => $license_id,
+            'users_id' => $user->getID(),
+        ])->getID();
+
+        // Count should still be 1 for Item_SoftwareLicense
+        $this->assertEquals(1, Item_SoftwareLicense::countForLicense($license_id));
+
+        // User count should be 1
+        $this->assertEquals(1, \SoftwareLicense_User::countForLicense($license_id));
+
+        // Total count should include both computer and user
+        $total_count = Item_SoftwareLicense::countForLicense($license_id) +
+            \SoftwareLicense_User::countForLicense($license_id);
+        $this->assertEquals(2, $total_count);
+
+        // Add another computer - this should be allowed as we're at the limit but not over
+        $computer2 = getItemByTypeName('Computer', '_test_pc02');
+        $this->createItem(Item_SoftwareLicense::class, [
+            'softwarelicenses_id' => $license_id,
+            'items_id' => $computer2->getID(),
+            'itemtype' => 'Computer',
+        ]);
+
+        // Cleanup
+        $user_license = new \SoftwareLicense_User();
+        $user_license->delete(['id' => $user_license_id]);
+
+        $item_license = new Item_SoftwareLicense();
+        $item_license->delete(['id' => $item_license_id]);
+
+        $software = new \Software();
+        $software->delete(['id' => $software_id]);
```

Comment:
```suggestion
```

You don't need to cleanup as the tests are executed in a DB transaction so the data is discarded after each tests.